File: custom_report/report/accounts/accounts.py
# Copyright (c) 2024, Ahmed Reda and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters=None):
	# columns, data = [], []
	columns =[
	_("parent_account") + "::200",
	_("account_name") + "::200"
	]


	parent_account = filters.get("parent_account")
	account_name = filters.get("account_name")

	sql =""" 
			SELECT 
				`tabAccount`.`parent_account`,
				`tabAccount`.`account_name`
			FROM `tabAccount`
			WHERE 
				(%(parent_account)s IS NULL OR `tabAccount`.`parent_account` Like %(parent_account)s)
				AND (%(account_name)s IS NULL OR `tabAccount`.`account_name` Like %(account_name)s)
			

	
	"""
	mydata= frappe.db.sql(sql,{
							"parent_account": f"%{parent_account}%" if parent_account else None,
							"account_name": f"%{account_name}%" if account_name else None,
							'indent': 0, 
							'has_value': True,
								},as_dict=True)
	return columns, mydata

File: custom_report/report/accounts/accounts.js
// Copyright (c) 2024, Ahmed Reda and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Accounts"] = {
    "filters": [
        {
            fieldname: "parent_account",
            label: "Please Select Parent Account",
            fieldtype: "Link",
            options: "Account",
            width: "350px"
        },
        {
            fieldname: "account_name",
            label: "Please Select Account Name",
            fieldtype: "Link",
            options: "Account",
            width: "350px"
        }
    ],

    "formatter": function (value, row, column, data, default_formatter) {
        if (column.fieldname == "account_name") {
            value = data.account_name;
            column.is_tree = true;  // Indicate that this column is part of a tree structure

            // Optionally, set an onclick event for opening a detailed view if necessary
            column.link_onclick =
                "frappe.query_reports['Accounts'].open_profit_and_loss_statement(" +
                JSON.stringify(data) +
                ")";
        }

        value = default_formatter(value, row, column, data);

        if (!data.parent_account) {
            var $value = $("<span>").text(value).css("font-weight", "bold");
            if (data.warn_if_negative && data[column.fieldname] < 0) {
                $value.addClass("text-danger");
            }

            value = $value.wrap("<p></p>").parent().html();
			console.log("VALUE",value);
			console.log("data",data);
			console.log("column",column);
			console.log("rows",row);
			console.log("default_formatter",default_formatter);
        }

        return value;
    },

    "tree": true,
    "name_field": "account_name",
    "parent_field": "parent_account",
    "initial_depth": 3,
    

};

